hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUnfreezeAccountHandler.java

```diff
@@ -117,6 +117,10 @@ private TokenRelation validateSemantics(
             throws HandleException {
         // Check that the token meta exists
         final var tokenId = op.tokenOrElse(TokenID.DEFAULT);
+
+        // Validate token is paused or deleted
+        TokenHandlerHelper.getIfUsable(tokenId, tokenStore);
```

Comment:
This seems correct. Just curious, were any tests failing because this line wasn't here?

Reply:
There were failing tests because of a missing validation for `TOKEN_IS_PAUSED`, but it made sense to use this utility function, because it also checks the `tokenId` and the `deleted` flag.

Reply:
Yes, for sure. We should probably write a unit test checking for a not found token ID, as well as a test for a deleted token

Reply:
Yes, agreed. Should I create issues for the new unit tests?